tests/utils.py

```diff
@@ -69,3 +70,8 @@ def format_expected_progress(progress):
         for sym in to_replace:
             progress = progress.replace(sym, to_replace[sym])
     return progress
+
+
+def clean_progress_output(output):
+    """Return output cleaned from \r, \n, and ANSI escape sequences"""
+    return re.sub('\n|\r|(\x1b\[|\x9b)[^@-_]*[@-_]|\x1b[@-_]', '', output)
```

Comment:
the regex is hard to follow. can you visually split it and add comments using https://docs.python.org/2/library/re.html#re.X ?